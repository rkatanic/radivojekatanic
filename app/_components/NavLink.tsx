"use client";

import classNames from "classnames";
import Link from "next/link";
import { usePathname } from "next/navigation";
import { twMerge } from "tailwind-merge";

interface Props {
  href: string;
  label: string;
}

const NavLink = ({ href, label }: Props) => {
  const pathname = usePathname();
  const isActive = pathname === href;

  return (
    <Link
      href={href}
      className={twMerge(
        classNames(
          "font-medium dark:text-zinc-400 dark:hover:text-zinc-300",
          isActive && "dark:text-zinc-100",
        ),
      )}
    >
      {label}
    </Link>
  );
};

export default NavLink;
