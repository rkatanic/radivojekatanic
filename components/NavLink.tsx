import Link from "next/link";
import { useRouter } from "next/router";

interface Props {
  link: string;
  label: string;
  icon: JSX.Element;
}

const NavLink = ({ link, label, icon }: Props): JSX.Element => {
  const { asPath } = useRouter();
  const isActiveLink = asPath.includes(link);

  return (
    <Link href={link}>
      <span
        className={`${
          isActiveLink
            ? "bg-sky-500 dark:text-gray-100"
            : "dark:text-gray-400 dark:hover:text-gray-200"
        } flex items-center gap-3 py-2 px-3 font-medium`}
      >
        {icon}
        {label}
      </span>
    </Link>
  );
};

export default NavLink;
