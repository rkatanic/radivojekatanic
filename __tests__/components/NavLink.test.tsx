import NavLink from "@/components/NavLink";
import { render } from "@testing-library/react";

const useRouter = jest.spyOn(require("next/router"), "useRouter");

describe("NavLink", () => {
  it("should render", (): void => {
    useRouter.mockImplementationOnce(() => ({
      asPath: "/",
    }));
    const { baseElement } = render(<NavLink link="/link" label="Label" />);

    expect(baseElement).toMatchSnapshot();
  });

  it("should render with active link", (): void => {
    useRouter.mockImplementationOnce(() => ({
      asPath: "/link",
    }));
    const { baseElement } = render(<NavLink link="/link" label="Label" />);

    expect(baseElement).toMatchSnapshot();
  });
});
