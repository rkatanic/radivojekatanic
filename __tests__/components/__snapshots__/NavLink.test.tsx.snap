// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink should render with active link 1`] = `
<body>
  <div>
    <span
      class="bg-sky-500 flex items-center gap-3 py-1.5 px-3 font-medium dark:text-white"
    >
      <div>
        icon
      </div>
      Label
    </span>
  </div>
</body>
`;

exports[`NavLink should render with inactive link 1`] = `
<body>
  <div>
    <span
      class="dark:text-gray-300 dark:hover:text-gray-200 flex items-center gap-3 py-1.5 px-3 font-medium dark:text-white"
    >
      <div>
        icon
      </div>
      Label
    </span>
  </div>
</body>
`;
